### PR TITLE
completed sgt-back-end

### PR DESCRIPTION
**Pull request by Dustin-Davis**
Screenshot 2022-01-28 130403
Screenshot 2022-01-28 152425
Screenshot 2022-01-28 160251
Screenshot 2022-01-28 161944

### CODE REVIEW

File: sgt-back-end/index.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const body = req.body;
+  const score = Number(req.body.score);
+  if (!body.name || !body.course || !body.score) {
+    res.status(400).json({ error: 'Entry must contain "Name", "Course", and "Score"' });
+    return;
+  } else if (!score || score < 0 || score > 100 || !Number.isInteger(score)) {
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:38]: `0` should be a valid `score`:

---

image

File: sgt-back-end/index.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const body = req.body;
+  const id = Number(req.params.gradeId);
+  const score = Number(req.body.score);
+  if (!id) {
+    res.status(400).json({ error: 'Invalid Id' });
+    return;
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:66]: Fractional and negative `id`s should also trigger a `400 Bad Request`:

---

image

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by Dustin-Davis** [comment]
Screenshot 2022-01-28 164744

File: sgt-back-end/index.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+app.post('/api/grades', (req, res) => {
+  const body = req.body;
+  const score = Number(req.body.score);
+  if (!body.name || !body.course || !body.score) {
+    res.status(400).json({ error: 'Entry must contain "Name", "Course", and "Score"' });
+    return;
+  } else if (score > 100 || !Number.isInteger(score)) {
+    res.status(400).json({ error: 'Invalid integer' });
+    return;
+  }
```
Comment (TimelikeClosure) [inline sgt-back-end/index.js:41]: Negative `score`s should not be valid.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**